src/operator/controllers/intents_reconcilers/kafka_acls.go

```diff
@@ -190,6 +191,13 @@ func (r *KafkaACLReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 	}
 
 	var result ctrl.Result
+
+	intents.Status.UpToDate = false
+
+	defer func() {
+		intents.Status.UpToDate = true
```

Comment:
You should set UpToDate = true only if there is no error, so move it above the `return ctrl.Result{}, nil` line.
If you do it it the `defer func()` block, then it will happen anyway, upon return. If an error or panic occurred, then the resource is not up to date.

Also, we need to remember to update the intents resource for this to take effect.

Reply:
Agreed, i will move the call after we have settled on what to do in the other conversation